exporter/tanzuobservabilityexporter/metrics_test.go

```diff
@@ -197,6 +196,223 @@ func TestGaugeConsumerMissingValue(t *testing.T) {
+		map[string]interface{}{
+			"env": "dev",
+		},
+		dataPoints,
+	)
+	sender := &mockSumSender{}
+	consumer := newSumConsumer(sender, nil)
+	assert.Equal(t, pdata.MetricDataTypeSum, consumer.Type())
+	var errs []error
+
+	// cumulative sums get treated as regular wavefront metrics
+	consumer.Consume(cumulativeMetric, &errs)
+	consumer.PushInternalMetrics(&errs)
+
+	expected := []tobsMetric{
+		{
+			Name:  "test.cumulative.metric",
+			Value: 62.25,
+			Ts:    1634205001,
+			Tags:  map[string]string{"env": "dev"},
+		},
+	}
+	assert.ElementsMatch(t, expected, sender.metrics)
+	assert.Empty(t, errs)
+}
+
+func TestSumConsumerUnspecified(t *testing.T) {
+	cumulativeMetric := newMetric(
+		"test.unspecified.metric", pdata.MetricDataTypeSum)
+	sum := cumulativeMetric.Sum()
+	sum.SetAggregationTemporality(pdata.MetricAggregationTemporalityUnspecified)
+	dataPoints := sum.DataPoints()
+	dataPoints.EnsureCapacity(1)
+	addDataPoint(
+		72.25,
+		1634206001,
+		map[string]interface{}{
+			"env": "qa",
+		},
+		dataPoints,
+	)
+	sender := &mockSumSender{}
+	consumer := newSumConsumer(sender, nil)
+	assert.Equal(t, pdata.MetricDataTypeSum, consumer.Type())
+	var errs []error
+
+	// unspecified sums get treated as regular wavefront metrics
+	consumer.Consume(cumulativeMetric, &errs)
+	consumer.PushInternalMetrics(&errs)
+
+	expected := []tobsMetric{
+		{
+			Name:  "test.unspecified.metric",
+			Value: 72.25,
+			Ts:    1634206001,
+			Tags:  map[string]string{"env": "qa"},
+		},
+	}
+	assert.ElementsMatch(t, expected, sender.metrics)
+	assert.Empty(t, errs)
+}
+
+func TestSumConsumerMissingValueNoLogging(t *testing.T) {
+	metric := newMetric("bad.metric", pdata.MetricDataTypeSum)
+	dataPoints := metric.Sum().DataPoints()
+	dataPoints.EnsureCapacity(1)
+	addDataPoint(
+		nil,
+		1633123456,
+		nil,
+		dataPoints,
+	)
+	sender := &mockSumSender{}
+	consumer := newSumConsumer(sender, nil)
+	var errs []error
+
+	consumer.Consume(metric, &errs)
+	consumer.PushInternalMetrics(&errs)
+
+	assert.Len(t, errs, 0)
+	assert.Empty(t, sender.metrics)
+	assert.Empty(t, sender.deltaMetrics)
+}
+
+func TestSumConsumerMissingValue(t *testing.T) {
+	metric := newMetric("bad.metric", pdata.MetricDataTypeSum)
+	sum := metric.Sum()
+	sum.SetAggregationTemporality(pdata.MetricAggregationTemporalityDelta)
+	dataPoints := sum.DataPoints()
+	dataPoints.EnsureCapacity(1)
+	addDataPoint(
+		nil,
+		1633123456,
+		nil,
+		dataPoints,
+	)
+	sender := &mockSumSender{}
+	observedZapCore, observedLogs := observer.New(zap.DebugLevel)
+	consumer := newSumConsumer(sender, &consumerOptions{
+		Logger:                zap.New(observedZapCore),
+		ReportInternalMetrics: true,
+	})
+	var errs []error
+
+	expectedMissingValueCount := 2
+	for i := 0; i < expectedMissingValueCount; i++ {
+		consumer.Consume(metric, &errs)
+	}
+	consumer.PushInternalMetrics(&errs)
+
+	assert.Len(t, errs, 0)
+	assert.Empty(t, sender.deltaMetrics)
+	require.Len(t, sender.metrics, 1)
+	assert.Equal(t, tobsMetric{
```

Comment:
Is `require.Len()` necessary if we change the assertion below it to `assert.ElementsMatch()`?

Reply:
Replaced with assert.Contains() which is what I do in the histogram tests.  Later we may publish other internal metrics. What we want to do in this test is make sure that of all the internal metrics pushed, they contain this missing value metric.